src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -100,6 +100,13 @@ public static function createResourceForClass(\ReflectionClass $reflectionClass)
      */
     protected function processValue($value, $isRoot = false)
     {
+        if ($value instanceof Reference && !$this->container->has($id = (string) $value)) {
+            if (preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\\\\[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)++$/', $id) && $reflectionClass = $this->container->getReflectionClass($id, true)) {
```

Comment:
This regex is used in subtly different variants in the code. I don't think it makes much sense to factorize it. There is nothing to fix in a uniform way across the code base, each place may require a different fix, if any.